curp/src/server/bg_tasks.rs

```diff
@@ -265,68 +306,91 @@ async fn bg_heartbeat<C: Command + 'static, Conn: ConnectInterface>(
         while !state.read().is_leader() {
             role_trigger.listen().await;
         }
+
+        let rpcs = state.map_read(|state_r| {
```

Comment:
Why did you extract this logic from `send_heartbeat` and place it here?  `send_heartbeat` throws the intention into relief.

Reply:
Two benefits:
1. Fix a bug: the leader should stop handling other heartbeats when one heartbeat failed because of term update(the leader will retire and a follower should not handle heartbeat responses).
2. `handle_heartbeat_responses` is easier for testing

Reply:
Sure. Refactored